packages/common/src/utils/coinUtils.ts

```diff
@@ -0,0 +1,33 @@
+import { UserCoin } from '~/api'
+
+/**
+ * Filters user coins based on the ARTIST_COINS feature flag and balance requirements
+ *
+ * @param userCoins - Array of user coins to filter
+ * @param isArtistCoinsEnabled - Whether the ARTIST_COINS feature flag is enabled
+ * @param wAudioMintAddress - The WAUDIO mint address from environment
+ * @returns Filtered array of user coins
+ */
+export const filterUserCoins = (
```

Comment:
nit: if you wanna get fancy, could make this a function that you pipe into a `userCoins.filter(filterUserCoins)` instead of what it is now, which accepts `userCoins`, filters it, and then spits it back out.

i <3 functional programming